### PR TITLE
SQL-1285: Remove ANSI endpoint

### PR DESCRIPTION
**Pull request by nbagnard**
Where to find the list of W functions => https://learn.microsoft.com/en-us/sql/odbc/reference/develop-app/unicode-function-arguments

### CODE REVIEW

**Review by terakilobyte — APPROVED** [review]
Changes LGTM. Looks like clippy and rustfmt are being tyrants.

File: odbc/src/api/diag.rs [hunk 1]
```diff
@@ -40,34 +25,6 @@ pub unsafe fn set_sql_statew(sql_state: &str, output_ptr: *mut WideChar) {
...
 }
 
-///
-/// get_diag_rec copies the given ODBC error's diagnostic information
-/// into the provided pointers.
-///
-/// # Safety
-/// This writes to multiple raw C-pointers
-///
-pub unsafe fn get_diag_rec(
```
Comment (jchemburkar) [inline odbc/src/api/diag.rs:50]: This looks good! I think we can simplify the `get_diag_field` function now as well to remove the `is_wstring` flag

File: odbc/src/api/functions.rs [hunk 2]
```diff
@@ -663,77 +602,6 @@ pub unsafe extern "C" fn SQLColumnPrivilegesW(
...
 
-///
-/// [`SQLColumns`]: https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/SQLColumns-function
-///
-/// # Safety
-/// Because this is a C-interface, this is necessarily unsafe
-///
-#[named]
-#[no_mangle]
-pub unsafe extern "C" fn SQLColumns(
```
Comment (jchemburkar) [inline odbc/src/api/functions.rs:674]: For these, and the other functions that _were_ implemented and now removed, should we add them to the list of `Unsupported Operators` in the README?

**Review by jchemburkar — APPROVED** [review]
LGTM! Just had a couple small thoughts

File: odbc/src/api/functions.rs (hunk 2 above)
Comment (nbagnard) [reply odbc/src/api/functions.rs:674]: Good call!
I looked at the README and I don't really see a need to go into this level of details. I'll add a line stating that the driver does not support ANSI endpoints, that should be enough, no need to list all the ANSI functions.
But, I still need to also clean-up the list of unsupported functions.

**Comment by nbagnard** [comment]
@jchemburkar, @terakilobyte : I updated the README, do you mind giving it one last look? I re-requested your review to make sure that it makes sense to you. I think it's simpler to state that we don't support ANSI functions and then list the ones that we don't support at all. But let me know if that is not working for you. I also cleaned the README and removed the 32-bit steps because we don't support 32-bit.
I left the registry update scripts because if people are coming to this page, they won't have access to the installer.

File: odbc/src/api/diag.rs [hunk 3]
```diff
@@ -150,41 +106,25 @@ pub unsafe fn get_diag_field(
...
-                            buffer_length as usize,
-                            string_length_ptr,
-                        ),
-                    },
+                    DiagType::SQL_DIAG_SQLSTATE => i16_len::set_output_wstring_as_bytes(
+                        error.get_sql_state(),
+                        diag_info_ptr,
+                        buffer_length as usize,
+                        string_length_ptr,
+                    ),
```
Comment (terakilobyte) [inline odbc/src/api/diag.rs:114]: Not related to this PR at all, but I'd really like refactoring all the string writing to cstr. I could create a ticket. WDYT?

**Review by terakilobyte — APPROVED** [review]
LGTM. Left one small "correction", no need to re-tag if you accept it!

File: odbc/src/api/diag.rs (hunk 3 above)
Comment (nbagnard) [reply odbc/src/api/diag.rs:114]: > Not related to this PR at all, but I'd really like refactoring all the string writing to cstr. I could create a ticket. WDYT?

@terakilobyte Please do :)

**Review by jchemburkar — APPROVED** [review]
The updates to the README look good to me!

**Comment by nbagnard** [comment]
evergreen merge

**Pull request merged**